File: JPAFlavorScape/src/main/java/com/skilldistillery/flavorscapeapp/entities/MenuItem.java
package com.skilldistillery.flavorscapeapp.entities;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;

@Entity
@Table (name = "menu_item")
public class MenuItem {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int Id; 
	@Column (name ="item_name")
	private  String itemName; 
	@Column (name ="item_url")
	private  String itemUrl; 
	private  Double price; 
	private  String description;
	private  int calories;
	private boolean enabled;
	public MenuItem() {
		super();
	}
	public int getId() {
		return Id;
	}
	public void setId(int id) {
		Id = id;
	}
	public String getItemName() {
		return itemName;
	}
	public void setItemName(String itemName) {
		this.itemName = itemName;
	}
	public String getItemUrl() {
		return itemUrl;
	}
	public void setItemUrl(String itemUrl) {
		this.itemUrl = itemUrl;
	}
	public Double getPrice() {
		return price;
	}
	public void setPrice(Double price) {
		this.price = price;
	}
	public String getDescription() {
		return description;
	}
	public void setDescription(String description) {
		this.description = description;
	}
	public int getCalories() {
		return calories;
	}
	public void setCalories(int calories) {
		this.calories = calories;
	}
	public boolean isEnabled() {
		return enabled;
	}
	public void setEnabled(boolean enabled) {
		this.enabled = enabled;
	}
	@Override
	public int hashCode() {
		return Objects.hash(Id);
	}
	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		MenuItem other = (MenuItem) obj;
		return Id == other.Id;
	}
	@Override
	public String toString() {
		return "MenuItem [Id=" + Id + ", itemName=" + itemName + ", itemUrl=" + itemUrl + ", price=" + price
				+ ", description=" + description + ", calories=" + calories + ", enabled=" + enabled + "]";
	} 
	
	
}

File: ngFlavorscape/src/app/models/menu-item.ts
import { Cuisine } from "./cuisine";
import { Menu } from "./menu";

export class MenuItem {
  id: number;
  itemName: String;
  enabled: boolean;
  itemUrl: string;
  description: string;
  calories: number;
  price: number;
  cuisines: Cuisine [];
  menu: Menu;

  constructor(
    id: number = 0,
    itemName: string = '',
    enabled: boolean = false,
    itemUrl: string = '',
    description: string = '',
    calories: number = 0,
    price: number = 0,
    cuisines: Cuisine [] = [],
    menu: Menu = new Menu()
  ){
    this.id = id;
    this.itemName = itemName;
    this.enabled = enabled;
    this.itemUrl = itemUrl;
    this.description = description;
    this.calories = calories;
    this.price = price;
    this.cuisines = cuisines;
    this.menu = menu;
  }
}
